src/json.rs

```diff
@@ -199,6 +199,42 @@
 //!     let decoded: TestStruct = json::decode(json_str.as_slice()).unwrap();
 //! }
 //! ```
+//!
+//! ## Parsing a `str` to `Json` and reading the result
+//!
+//! ```rust
+//! # #![allow(unstable)]
+//! extern crate "rustc-serialize" as rustc_serialize;
+//! use rustc_serialize::json::Json;
+//!
+//! fn main() {
+//!     let data = Json::from_str("{\"foo\": 13, \"bar\": \"baz\"}").unwrap();
+//!     println!("data: {}", data);
+//!     // data: {"bar":"baz","foo":13}
+//!     println!("object? {}", data.is_object());
+//!     // object? true
+//!
+//!     let obj = data.as_object().unwrap();
+//!     let foo = obj.get("foo").unwrap();
+//!     println!("array? {}", foo.is_array());
```

Comment:
Maybe this and the line below could just only use `as_`, e.g.

```rust
println!("array? {:?}", foo.as_array());
// array? None
println!("u64? {:?}", foo.as_u64());
// u64? Some(13u64)
```

I'm not totally sold on it, but it's generally nicer to use the `Option`-returning one to both check for validity and get the value, rather than pre-checking with an `is` method (more type guarantees).

What do you think?